phoenix-core-client/src/main/java/org/apache/phoenix/index/IndexMaintainer.java

```diff
@@ -1789,6 +1788,9 @@ public static IndexMaintainer fromProto(ServerCachingProtos.IndexMaintainer prot
         } else {
             maintainer.isCDCIndex = false;
         }
+        if (proto.hasDataTableSaltBuckets()) {
+            maintainer.nDataTableSaltBuckets = proto.getDataTableSaltBuckets();
```

Comment:
If the proto doesn't have data table salt buckets then what is the nDataTableSaltBuckets field ? I think it will be 0 not -1. Seems like -1 is constant that is introduced is unnecessary.

Reply:
Agree on the inconsistency, but I think it should be set to `-1` to be consistent than `0` for the same reason I [mentioned](https://github.com/apache/phoenix/pull/1969/files#r1770766770) in the other thread.